### PR TITLE
refactor: more utility functions

### PR DESCRIPTION
**Pull request by nistei**
## Summary of Changes
Add utilities for:
- Installation detection
- Installation cleanup
- MSFS detection

## Screenshots (if necessary)
N/A

## Additional context
N/A

Discord username (if different from GitHub): nistei#1362

### CODE REVIEW

File: src/renderer/utils/Msfs.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import net from "net";
+
+export class Msfs {
+    static async isRunning(): Promise<boolean> {
```
Comment (Benjozork) [inline src/renderer/utils/Msfs.ts:4]:
```suggestion
export class Msfs {

    static async isRunning(): Promise<boolean> {
```

File: src/renderer/utils/Msfs.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                resolve(true);
+                socket.destroy();
+            });
+            socket.on('error', () => {
+                resolve(false);
+                socket.destroy();
+            });
+        });
+    }
+}
```
Comment (Benjozork) [inline src/renderer/utils/Msfs.ts:18]:
```suggestion
    }
}
```
```suggestion
    }
    
}
```

File: src/renderer/utils/Directories.ts [hunk 3]
```diff
@@ -13,4 +15,44 @@ export class Directories {
...
+            const symlinkPath = fs.readlinkSync(targetDir);
+            if (symlinkPath && fs.existsSync(path.join(symlinkPath, '/../.git'))) {
+                console.log('Is git repo', targetDir);
+                return true;
+            }
+        } catch {
+            console.log('Is not git repo', targetDir);
+            return false;
+        }
+    }
```
Comment (Benjozork) [inline src/renderer/utils/Directories.ts:57]:
```suggestion
        }
        
    }
```

**Review by Benjozork — APPROVED** [review]
LMGT

**Review by FoxtrotSierra6829 — APPROVED** [review]
fix the lint pls, thereafter LGTM

**Review by FoxtrotSierra6829 — APPROVED** [review]

**Pull request merged**